main.go

```diff
@@ -269,77 +266,63 @@ func validate(in configInput) error {
 }
 
 func generate(in configInput) ([]byte, error) {
-	xdsServer := server{
-		ServerUri:    in.xdsServerUri,
-		ChannelCreds: []creds{{Type: "google_default"}},
-	}
+	c := &config{}
```

Comment:
This is harder to read and verify. I really can't easily tell what is in the config except by looking at the test results. I don't see the need to rewrite this function when removing the `includeXDSTPNameInLDS` if, when all we needed to do was inline its contents into the `config` literal.